libs/viewer/src/SimpleViewer.cpp

```diff
@@ -783,6 +797,177 @@ void SimpleViewer::changeAllVisibility(utils::Entity entity, bool changeToVisibl
+    "SRGB8_ALPHA8_ASTC_10x8",
+    "SRGB8_ALPHA8_ASTC_10x10",
+    "SRGB8_ALPHA8_ASTC_12x10",
+    "SRGB8_ALPHA8_ASTC_12x12",
+};
+
+
+std::string SimpleViewer::validateTweaks(const TweakableMaterial& tweaks) {
+    auto verifyTextured = [&]<typename T, bool MayContainFile, bool IsColor, bool IsDerivable>(std::string& result, const std::string& prompt, const TweakableProperty<T, MayContainFile, IsColor, IsDerivable>& prop, filament::Texture::InternalFormat expectedFormat = filament::Texture::InternalFormat::UNUSED) {
+        if (prop.isFile) {
+            if (prop.filename.getFileExtension() != "png") {
+                result += "ERROR: " + prompt + " is not a png.\n";
+            }
+
+            auto textureEntry = mTextures.find(prop.filename.asString());
+            if (textureEntry != mTextures.end()) {
+                int expectedChannelCount;
+                // Infer the expected texture format if no explicit cue was given by the caller
+                if (expectedFormat == filament::Texture::InternalFormat::UNUSED) {
+                    if (IsColor) {
+                        if (tweaks.mShaderType == TweakableMaterial::MaterialType::TransparentSolid) {
+                            expectedFormat = filament::Texture::InternalFormat::SRGB8_A8;
+                            expectedChannelCount = 4;
+                        } else {
+                            expectedFormat = filament::Texture::InternalFormat::SRGB8;
+                            expectedChannelCount = 3;
+                        }
+                    } else {
+                        expectedFormat = filament::Texture::InternalFormat::R8;
+                        expectedChannelCount = 1;
+                    }
+                } else {
+                    if (expectedFormat == filament::Texture::InternalFormat::SRGB8_A8) expectedChannelCount = 4;
+                    else if (expectedFormat == filament::Texture::InternalFormat::SRGB8 || expectedFormat == filament::Texture::InternalFormat::RGB8) expectedChannelCount = 3;
+                    else if (expectedFormat == filament::Texture::InternalFormat::R8 ) expectedChannelCount = 1;
+                }
+
+                if (textureEntry->second->getFormat() != expectedFormat) {
+                    result += "ERROR: " + prompt + " has incorrect format! Expected " + formatToName[(int16_t)expectedFormat] + ", got " + formatToName[(int16_t)textureEntry->second->getFormat()] + ".\n";
+                }
+
+                auto textureChannelCountEntry = mTextureFileChannels.find(prop.filename.asString());
+                if (textureChannelCountEntry != mTextureFileChannels.end()) {
+                    if (textureChannelCountEntry->second != expectedChannelCount) {
+                        result += "ERROR: " + prompt + " has incorrect number of channels! Expected " + std::to_string(expectedChannelCount) + ", got " + std::to_string(mTextureFileChannels[prop.filename.asString()]) + ".\n";
```

Comment:
nit: Actual channel count could be `textureChannelCountEntry->second` instead of `mTextureFileChannels[prop.filename.asString()]`

Reply:
Fixed in https://github.com/shapr3d/filament/pull/20/commits/ddf94a75580b2a0573fd9d18dc5bc78c44e25463